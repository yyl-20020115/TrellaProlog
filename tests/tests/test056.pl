:- initialization(main).

sentence --> np, vp.
np --> det, noun.
vp --> verb, np.
vp --> verb.

noun --> [woman].
noun --> [man].
verb --> [shoots].
det --> [the].
det --> [a].

/*
    Generate all possible sentences...
*/

main :- phrase(sentence, X), write(X), nl, fail.
main.
