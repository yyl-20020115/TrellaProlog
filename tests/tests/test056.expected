[the,woman,shoots,the,woman]
[the,woman,shoots,the,man]
[the,woman,shoots,a,woman]
[the,woman,shoots,a,man]
[the,woman,shoots]
[the,man,shoots,the,woman]
[the,man,shoots,the,man]
[the,man,shoots,a,woman]
[the,man,shoots,a,man]
[the,man,shoots]
[a,woman,shoots,the,woman]
[a,woman,shoots,the,man]
[a,woman,shoots,a,woman]
[a,woman,shoots,a,man]
[a,woman,shoots]
[a,man,shoots,the,woman]
[a,man,shoots,the,man]
[a,man,shoots,a,woman]
[a,man,shoots,a,man]
[a,man,shoots]
